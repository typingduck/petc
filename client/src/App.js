import React from 'react'
import { Switch, Route, withRouter } from 'react-router-dom'

import './App.css'

import HomePage from './HomePage'

function App () {
  return (
    <Switch>
      <Route exact path='/' component={HomePage} />
      <Route path='/docs:number' component={React.Component} />
    </Switch>
  )
}

export default withRouter(App)
