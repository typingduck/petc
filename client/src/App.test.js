/* global expect it */
import React from 'react'
import Renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'

import App from './App'

it('renders homepage', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <App />
    </MemoryRouter>
  )
  expect(component.toJSON()).toMatchSnapshot()
})

it('renders doc', () => {
  const component = Renderer.create(
    <MemoryRouter initialEntries={['/docs/123']}>
      <App />
    </MemoryRouter>
  )
  expect(component.toJSON()).toMatchSnapshot()
})
