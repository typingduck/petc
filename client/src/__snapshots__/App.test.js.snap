// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders doc 1`] = `null`;

exports[`renders homepage 1`] = `
<div
  className="petc-homepage"
>
  <a
    href="#new"
    onClick={[Function]}
  >
    Create New
  </a>
</div>
`;
